qiskit_experiments/analysis/curve_analysis.py

```diff
@@ -939,10 +940,27 @@ def _run_analysis(
         # pop arguments that are not given to fitter
         extra_options = self._arg_parse(**options)
 
-        # get experiment metadata
+        # get experiment metadata and run-options dependent data processor
         try:
             self.__experiment_metadata = experiment_data.metadata()
-        except AttributeError:
+
+            # No data processor has been provided at run-time we infer one from the job
+            # metadata and default to the data processor for averaged classified data.
+            if getattr(self, "__data_processor") is None:
+                try:
+                    run_options = self.__experiment_metadata["job_metadata"][-1]["run_options"]
+                except IndexError:
+                    run_options = {}
+
+                meas_level = run_options.get("meas_level", MeasLevel.CLASSIFIED)
+                meas_return = run_options.get("meas_return", "avg")
+                normalization = self._get_option("normalization")
+
+                processor = get_to_signal_processor(meas_level, meas_return, normalization)
```

Comment:
This should be probability processor. Most of experiments are run with level2 and we need level2 probability. IQ signal is too limited use case to be assumed in the base class.

Reply:
This will return a probability processor if you ask for level 2 which is here the default. See here: https://github.com/Qiskit/qiskit-experiments/blob/112f4ce54865248c841973fc58564c975aac81df/qiskit_experiments/data_processing/processor_library.py#L40-L41

Reply:
Can you rename the getter function in the separate PR? It sounds bit confusing.

Reply:
Done here: 5e6a82a